geps/gep-1731/index.md

```diff
@@ -0,0 +1,438 @@
+# GEP-1731: HTTPRoute Retries
+
+* Issue: [#1731](https://github.com/kubernetes-sigs/gateway-api/issues/1731)
+* Status: Implementable
+
+(See status definitions [here](/geps/overview/#gep-states).)
+
+## TLDR
+
+To allow configuration of a Gateway to retry unsuccessful requests to backends before sending a response to a client request.
+
+## Goals
+
+* To allow specification of [HTTP status codes](https://www.rfc-editor.org/rfc/rfc9110#name-overview-of-status-codes) for which a request should be retried.
+* To allow specification of the maximum number of times to retry a request.
+* To allow specification of the minimum backoff interval between retry attempts.
+* To define any interaction with configured HTTPRoute [timeouts](/geps/gep-1742/).
+* Retry configuration must be applicable to most known Gateway API implementations.
+
+## Future Goals
+
+* To allow specification of a retry ["budget"](https://finagle.github.io/blog/2016/02/08/retry-budgets/) to determine whether a request should be retried, and any shared configuration or interaction with max count retry configuration.
+* Define more precise semantics for retry configuration on "consumer" vs "producer" routes for mesh implementations.
+
+## Non-Goals
+
+* To allow more granular control of the backoff strategy than many dataplanes allow customizing, such as whether to use an exponential backoff interval between retry attempts, add jitter, or cap the backoff interval to a maximum duration.
+* To allow specification of a default retry policy for all routes in a given namespace or attached to a particular Gateway.
+* A standard API for approaches for retry logic other than max count or "budget", such as interaction with rate limiting headers.
+* To allow specification of gRPC status codes for which a request should be retried (this should be covered in a separate GEP).
+* Support for unary or bidirectional streams, which may have different considerations for timeouts or request/response patterns within the stream after establishment (this should be a separate GEP).
```

Comment:
This language is odd to me (I keep reading it as "unary streams and bidirectional streams") and this feels more like something we'll want in another GEP anyway. Maybe

```suggestion
* Support for streaming or bidirectional APIs (these could be covered by a future GEP).
```